pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -786,87 +871,65 @@ func imageRegistryCommands(imageId string, db *v1alpha1.DependencyBuild, jbsConf
 		"--scm-uri=" + db.Spec.ScmInfo.SCMURL,
 		"--scm-commit=" + db.Spec.ScmInfo.CommitHash,
 	}
-	konfluxArgs = append(konfluxArgs, gitArgs...)
+	konfluxArgs = append(konfluxArgs, gitArgs(jbsConfig, db)...)
 	konfluxArgs = append(konfluxArgs, "--image-id="+imageId)
 
-	hermeticPreBuildImageArgs := []string{
-		"deploy-hermetic-pre-build-image",
-		"--source-image=$(params." + PreBuildImageDigest + ")",
-		"--image-name=" + hermeticPreBuildImageTag,
-		"--build-artifact-path=$(workspaces.source.path)/artifacts",
-		"--image-source-path=" + MavenArtifactsPath,
-		"--repository-path=$(workspaces.source.path)/build-info/",
-		"--image-hash=$(results." + HermeticPreBuildImageDigest + ".path)",
-	}
-	hermeticPreBuildImageArgs = append(hermeticPreBuildImageArgs, registryArgs...)
-
-	return preBuildImageArgs, copyArtifactsArgs, deployArgs, konfluxArgs, hermeticDeployArgs, hermeticPreBuildImageArgs
-}
-
-func tagCommands(jbsConfig *v1alpha1.JBSConfig) []string {
-
-	tagArgs := []string{
-		"tag-container",
-		"--image-digest=$(params." + DeployedImageDigestParam + ")",
-	}
-	registryArgs := registryArgs(jbsConfig)
-	tagArgs = append(tagArgs, registryArgs...)
-	tagArgs = append(tagArgs, "$(params.GAVS)")
-
-	return tagArgs
+	return preBuildImageArgs, postBuildImageArgs, copyArtifactsArgs, deployArgs, konfluxArgs, hermeticDeployArgs, hermeticPreBuildImageArgs
 }
 
-func registryArgs(jbsConfig *v1alpha1.JBSConfig) []string {
+// This effectively duplicates the defaults from DeployPreBuildImageCommand.java
+func registryArgsWithDefaults(jbsConfig *v1alpha1.JBSConfig, preBuildImageTag string) string {
 
 	imageRegistry := jbsConfig.ImageRegistry()
-	registryArgs := make([]string, 0)
+	var registryArgs strings.Builder
 	if imageRegistry.Host != "" {
-		registryArgs = append(registryArgs, "--registry-host="+imageRegistry.Host)
+		registryArgs.WriteString(imageRegistry.Host)
+	} else {
+		registryArgs.WriteString("quay.io")
 	}
 	if imageRegistry.Port != "" && imageRegistry.Port != "443" {
-		registryArgs = append(registryArgs, "--registry-port="+imageRegistry.Port)
+		registryArgs.WriteString(":")
+		registryArgs.WriteString(imageRegistry.Port)
 	}
+	// 'else' :
+	// No need to pass a default port (of 443) and its not supported by select-oci-auth.sh according
+	// to the tests in https://github.com/konflux-ci/build-trusted-artifacts/pull/103
+	registryArgs.WriteString("/")
 	if imageRegistry.Owner != "" {
-		registryArgs = append(registryArgs, "--registry-owner="+imageRegistry.Owner)
+		registryArgs.WriteString(imageRegistry.Owner)
+	} else {
+		// TODO: Should we keep the owner defaulted as 'hacbs' (or even jbs) ? I think this is legacy and should be removed?
+		registryArgs.WriteString("hacbs")
```

Comment:
In the java code the owner was defaulted e.g.
```
    @CommandLine.Option(names = "--registry-owner", defaultValue = "hacbs")
    String owner;
```
Should we keep this or make it an error condition if the owner is not specified?

Reply:
We can remove this IMHO